tests/components/bsblan/conftest.py

```diff
@@ -78,3 +80,80 @@ async def init_integration(
     await hass.async_block_till_done()
 
     return mock_config_entry
+
+
+# ZeroconfServiceInfo fixtures for different discovery scenarios
+
+
+@pytest.fixture
+def zeroconf_discovery_info() -> ZeroconfServiceInfo:
+    """Return zeroconf discovery info for a BSBLAN device with MAC address."""
+    return ZeroconfServiceInfo(
+        ip_address=ip_address("10.0.2.60"),
+        ip_addresses=[ip_address("10.0.2.60")],
+        name="BSB-LAN web service._http._tcp.local.",
+        type="_http._tcp.local.",
+        properties={"mac": "00:80:41:19:69:90"},
+        port=80,
+        hostname="BSB-LAN.local.",
+    )
+
+
+@pytest.fixture
+def zeroconf_discovery_info_no_mac() -> Mock:
+    """Return zeroconf discovery info for a BSBLAN device without MAC address."""
+    discovery_info = Mock()
+    discovery_info.ip_address = ip_address("10.0.2.60")
+    discovery_info.ip_addresses = [ip_address("10.0.2.60")]
+    discovery_info.name = "BSB-LAN web service._http._tcp.local."
+    discovery_info.type = "_http._tcp.local."
+    discovery_info.properties = {}  # No MAC in properties
+    discovery_info.properties_raw = {}  # No MAC in properties_raw either
+    discovery_info.port = 80
+    discovery_info.hostname = "BSB-LAN.local."
+    return discovery_info
```

Comment:
why is this a mock?

Reply:
ZeroconfServiceInfo doesn’t support the properties_raw, which is why it’s a mock. Is there another way to achieve this? I suppose I need to modify the firmware to make the Mac Address just a property, but it’s not currently implemented that way.